bundles/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/handler/SpeedTestHandler.java

```diff
@@ -65,90 +63,84 @@ public SpeedTestHandler(Thing thing) {
 
     @Override
     public void initialize() {
-        configuration = this.getConfigAs(SpeedTestConfiguration.class);
-        startAutomaticRefresh();
-        updateStatus(ThingStatus.ONLINE);
-    }
-
-    private void startAutomaticRefresh() {
-        final int initialDelay = 2;
-        refreshTask = scheduler.scheduleWithFixedDelay(this::launchSpeedTest, initialDelay,
+        configuration = getConfigAs(SpeedTestConfiguration.class);
+        logger.info("Speedtests starts in {} minutes, then refreshes every {} minutes",
+                configuration.initialDelayMinutes, configuration.refreshInterval);
+        refreshTask = scheduler.scheduleWithFixedDelay(this::startSpeedTest, configuration.initialDelayMinutes,
                 configuration.refreshInterval, TimeUnit.MINUTES);
-        logger.info("Start automatic in {} minutes, refreshed every {} minutes", initialDelay,
-                configuration.refreshInterval);
+        updateStatus(ThingStatus.ONLINE);
     }
 
-    private void launchSpeedTest() {
-        updateState(CHANNEL_TEST_ISRUNNING, OnOffType.ON);
-        updateState(CHANNEL_TEST_START, new DateTimeType());
-        updateState(CHANNEL_TEST_END, UnDefType.NULL);
+    @SuppressWarnings("null")
+    synchronized private void startSpeedTest() {
         if (speedTestSocket == null) {
+            logger.debug("Network speedtest started");
             speedTestSocket = new SpeedTestSocket(1500);
```

Comment:
```suggestion
            final SpeedTestSocket socket = new SpeedTestSocket(1500);
            speedTestSocket = socket;
```

and replace the local occurences of `speedTestSocket` with `socket`. You can remove the null-warning-suppression then.

Reply:
Cleaner ! Thanks for the hint